### PR TITLE
HARP-5432 Normal vectors are now the projection's surface normal.

### PR DESCRIPTION
**Pull request by nzjony**

### CODE REVIEW

**Comment by nzjony** [comment]
Please also see the gerrit patchset.

**Review by CraigBerlin — COMMENTED** [review]
Terrain is not being displayed with these changes:

**Comment by nzjony** [comment]
@heremaps-bot retest this please
@CraigBerlin, yes, unfortunately I pushed the gerrit patchset as draft, and I think that is why this was built without the changes there.

File: @here/harp-datasource-protocol/lib/Techniques.ts [hunk 1]
```diff
@@ -319,11 +319,7 @@ export function isShaderTechnique(technique: Technique): technique is ShaderTech
  * @param technique Technique to check.
  */
 export function needsVertexNormals(technique: Technique): boolean {
-    return (
-        isStandardTechnique(technique) ||
-        isTerrainTechnique(technique) ||
-        isStandardExtrudedLineTechnique(technique)
-    );
+    return isStandardTechnique(technique) || isStandardExtrudedLineTechnique(technique);
```
Comment (ninok) [inline @here/harp-datasource-protocol/lib/Techniques.ts]: The code in Tile.ts is only calling this method if there are no normal yet, so no need for this change.
```
                if (!bufferGeometry.getAttribute("normal") && needsVertexNormals(technique)) {
                    bufferGeometry.computeVertexNormals();
                }
```

Comment (nzjony) [reply @here/harp-datasource-protocol/lib/Techniques.ts]: Agreed, https://github.com/heremaps/harp.gl/pull/475

**Pull request merged**